### PR TITLE
Automatic Software Update Resource

### PR DESCRIPTION
**Pull request by v-anshie**
---
name: Automatic Software Update Resource
about: Improve state management of software updates.
title: "Automatic Software Update Resource"
labels: ''
assignees: 'Andre Shields'

---

## Describe the Pull Request  
Add resource that manages the state of the desired software update preferences using the **plist** resource to set the values for each individual property.

Resolves AB#2670181

## Justification  
Use chef to enable or disable automatic software update preferences.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,9 +58,42 @@ to always keep macOS on and available.
 | `node['macos']['network_time_server']`  | `'time.windows.com'`    |
 | `node['macos']['time_zone']`            | `'America/Los_Angeles'` |
 
+### Xcode
```
Comment (ehanlon) [inline README.md]: This was removed in #187

File: README.md [hunk 2]
```diff
@@ -36,6 +36,14 @@ should be an admin user with passwordless super-user rights.
 
 ## Recipes
 
+### Disable Software Updates
```
Comment (ehanlon) [inline README.md]: This was removed in #188

File: README.md (hunk 1 above)
Comment (v-anshie) [reply README.md]: Changed by rebasing release/3.0 into this branch and reverting commit.

File: README.md (hunk 2 above)
Comment (v-anshie) [reply README.md]: Changed by rebasing release/3.0 into this branch and reverting commit.

File: spec/unit/resources/automatic_software_updates_spec.rb [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      .with(entry: 'AutomaticCheckEnabled',
+            value: true,
+            path: '/Library/Preferences/com.apple.SoftwareUpdate.plist')
+  }
+
+  it {
+    is_expected.to set_plist('entry for CriticalUpdateInstall')
+      .with(entry: 'CriticalUpdateInstall',
+            value: true,
+            path: '/Library/Preferences/com.apple.SoftwareUpdate.plist')
```
Comment (mjmerin) [inline spec/unit/resources/automatic_software_updates_spec.rb]:
```suggestion
            path: software_update_plist)
```

File: documentation/resource_automatic_software_updates.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**Ruby type:** `TrueClass, FalseClass`
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Enable or disable automatic install of critical updates.
+Only applicable if the `check` property is enabled
+
+`install_app_store`
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**Ruby type:** `TrueClass, FalseClass`
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Enable or disable automatic install of app store updates.
```
[the remark this answers is not in the document]
Comment (v-anshie) [reply documentation/resource_automatic_software_updates.md]: In the actual preference they refer to it as just app store. Do we still want to refer to it as Mac App Store in the documentation?

File: documentation/resource_automatic_software_updates.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+automatic_software_updates
+=====
+
+Use the **automatic_software_updates** resource to manage software update preferences.
```
Comment (americanhanko) [inline documentation/resource_automatic_software_updates.md]: This should say manage the Automatic Software Update preferences. Also, let's get an inline screenshot of what this looks like in System Preferences. Here's the screenshot for you:

image

File: documentation/resource_automatic_software_updates.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (americanhanko) [reply documentation/resource_automatic_software_updates.md]: macOS updates are no longer managed in the App Store – this should say "enable or disable install of macOS software updates"

File: documentation/resource_automatic_software_updates.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+Only applicable if the `download` property is `true`.
+
+Examples
+----------
+
+**Enable automatic checking of software updates**
+
+```ruby
+automatic_software_updates "enable automatic checking of software updates" do
+check true
```
Comment (americanhanko) [inline documentation/resource_automatic_software_updates.md]: Looks like we're missing some indentation here.

**Pull request merged**